gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -249,8 +249,8 @@ public void orchestrate(Spec spec, Properties jobProps, long triggerTimestampMil
       // If multi-active scheduler is enabled do not pass onto DagManager, otherwise scheduler forwards it directly
       // Skip flow compilation as well, since we recompile after receiving event from DagActionStoreChangeMonitor later
       if (flowTriggerHandler.isPresent()) {
-        // If triggerTimestampMillis is 0, then it was not set by the job trigger handler, and we cannot handle this event
-        if (triggerTimestampMillis == 0L) {
+        // If triggerTimestampMillis is -1, then it was not set by the job trigger handler, and we cannot handle this event
+        if (triggerTimestampMillis == -1L) {
```

Comment:
seems fine to adopt `-1` as a sentinel value for "never set".

(I'm presuming we're only doing it because it would be a more far reaching rework to use `Optional<Long>`, which would most clearly capture the situation.)

still, let's give this constant a name to describe its meaning.  e.g.`TRIGGER_EVENT_TIME_NEVER_SET`